scalameta/dialects/shared/src/main/scala/scala/meta/dialects/package.scala

```diff
@@ -144,10 +144,18 @@ package object dialects {
     .withAllowStarWildcardImport(true)
     .withAllowProcedureSyntax(false)
     .withAllowDoWhile(false)
-    .withAllowPlusMinusUnderscoreAsPlaceholder(true)
     .withUseInfixTypePrecedence(true)
     .withAllowInfixOperatorAfterNL(true)
 
+  implicit val Scala31 = Scala30.duplicate
+
+  implicit val Scala32 = Scala31
+    .withAllowPlusMinusUnderscoreAsPlaceholder(true)
```

Comment:
I think we should keep it enabled starting from scala30.

It might be enabled by `-Ykind-projector` in compiler and as I remember we decided to enable it by default to avoid introducing an additional dialect. This syntax anyway won't be parsed without this flag so it doesn't produce any any ambiguity.

Reply:
I'm planning to rename this flag and replace with the one which supports underscore as placeholder in types overall, not just covariant. and that one was added in 3.2.

Reply:
also, scala3 points to 33, so it's enabled for that dialect.

Reply:
Then maybe do dialect changes together with the changes you planned to do after it?
It will be a bit more clear what will be the difference between these dialects

Reply:
@dos65 it's all described in the linked #2807. the dialect field is renamed in #2811 (conflict between this pr and that one will be resolved based on which pr gets merged first).